ios/FlankerNativeComponents/GameManager.swift

```diff
@@ -333,27 +369,33 @@ class GameManager {
       delegate?.updateText(text: text, color: .black, font: Constants.bigFont, isStart: true, typeTime: .trial)
     }
 
+    setEndTimeViewingImage(time: CACurrentMediaTime(), isStart: true, type: .trial)
+
     timeResponse = Timer(timeInterval: gameParameters.trialDuration / 1000, target: self, selector: #selector(self.timeResponseFailed), userInfo: nil, repeats: false)
     RunLoop.main.add(timeResponse!, forMode: .common)
   }
 
   @objc func timeResponseFailed() {
+    invalidateTimers()
     guard let gameParameters = gameParameters else { return }
 
-    delegate?.setEnableButton(isEnable: false)
     if gameParameters.showFeedback {
+      delegate?.setEnableButton(isEnable: false)
       delegate?.updateText(text: Constants.timeRespondText, color: .black, font: Constants.smallFont, isStart: false, typeTime: .feedback)
     }
 
     guard
-      let startTrialTimestamp = startTrialTimestamp,
-      let endTrialTimestamp = endTrialTimestamp
+      let startTrialTimestamp = startTrialTimestamp
     else { return }
 
+    if endTrialTimestamp == nil {
+      endTrialTimestamp = CACurrentMediaTime()
+    }
```

Comment:
What behavior is changing where previously we always had a end timestamp (from the `guard let`) that we now may or may not have a timestamp.